libs/entitlement/src/main/java/org/elasticsearch/entitlement/instrumentation/MethodKey.java

```diff
@@ -16,6 +16,8 @@
  *
  * @param className      the "internal name" of the class: includes the package info, but with periods replaced by slashes
  * @param methodName     the method name
- * @param parameterTypes a list of "internal names" for the parameter types
+ * @param parameterTypes a list of "internal names" for the parameter types that appear in the method's descriptor (not the receiver)
+ * @param hasReceiver    true if the target method is a nonstatic method, in which case the receiver ({@code this})
+ *                       passed as an argument to the check method; false for static methods and constructors.
  */
-public record MethodKey(String className, String methodName, List<String> parameterTypes) {}
+public record MethodKey(String className, String methodName, List<String> parameterTypes, boolean hasReceiver) {}
```

Comment:
since the receiver is about the check method, perhaps `isInstance` makes more sense?

Reply:
Why did we re-introduce this? 
It is not used anywhere, so it's useless?